src/gui/mzroll/autoupdater.cpp

```diff
@@ -0,0 +1,124 @@
+#define _STR(X) #X
+#define STR(X) _STR(X)
+
+#include <QProcess>
+#include <QDebug>
+#include <QApplication>
+
+#include <pugixml.hpp>
+
+#include <common/downloadmanager.h>
+#include "autoupdater.h"
+#include "base64.h"
+#include "projectversioning.h"
+
+void AutoUpdater::startMaintenanceTool()
+{
+    // Windows: <install_dir>/El-MAVEN/bin/
+    // MacOS:   <install_dir>/El-MAVEN/bin/El-MAVEN.app/Contents/MacOS/
+    QDir applicationDir(QCoreApplication::applicationDirPath());
+
+#ifdef __OSX_AVAILABLE
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/El-MAVEN.app/Contents/
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/El-MAVEN.app/
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/bin/
+#endif
+    applicationDir.cdUp(); // <install_dir>/El-MAVEN/
+    applicationDir.cdUp(); // <install_dir>/
+
+    QString maintenanceToolPath = "";
+#ifdef __OSX_AVAILABLE
+    maintenanceToolPath = applicationDir.path()
+                          + QDir::separator()
+                          + "maintenancetool.app/Contents/MacOS/maintenancetool";
+#endif
+#ifdef Q_OS_WIN
+    maintenanceToolPath = applicationDir.path()
+                          + QDir::separator()
+                          + "maintenancetool.exe";
+#endif
+
+    qDebug() << "Using path"
+             << maintenanceToolPath
+             << "for maintenancetool executable.";
+    if (!QFile::exists(maintenanceToolPath)) {
+        qWarning() << "Maintenance tool was not found at its expected path. "
+                      "Cannot proceed with update process.";
+        return;
+    }
+
+    QProcess* maintenanceToolProcess = new QProcess;
+    connect(maintenanceToolProcess,
+            &QProcess::started,
+            this,
+            &AutoUpdater::maintenanceToolStarted);
+    connect(maintenanceToolProcess,
+            static_cast<void(QProcess::*)(int, QProcess::ExitStatus)>(&QProcess::finished),
+            this,
+            &AutoUpdater::maintenanceToolClosed);
+
+    maintenanceToolProcess->setProgram(maintenanceToolPath);
+    maintenanceToolProcess->setArguments({"--updater"});
```

Comment:
Why not the `silentUpdate` argument ?

Reply:
Because with `silentUpdate`, the updater during its init stage tries to (not sure why) run the control script (from packaging repo), which includes some custom GUI logic (for our privacy policy page). And this crashes the non-GUI tool before being able to even check for updates.

There must be a solution for this, which I am open to, but we are a bit tight on time here. Future releases can have behind the scenes auto-updates.

Reply:
@saifulbkhan 
Sounds familiar. And IIRC I did find a fix for it
I think the the logic inside the control script should be made to run only if the `MaintenanceTool` is running in the **Installer** mode(this is the online/offline installer that we prepare using our packaging script) and not in the **Updater** mode(this is when we run the `Maintenancetool` and choose the **Update mode** or when it's run using the **silentUpdate** argument)

Reply:
```
function Component()
{
    ...

    installer.setDefaultPageVisible(QInstaller.StartMenuSelection, false)
    
    if(installer.isInstaller()) {
        installer.addWizardPage(component, "PolicyPage", QInstaller.PerformInstallation)
        gui.pageByObjectName("DynamicPolicyPage").entered.connect(enteredPolicyPage)
    }

   ...
}
```
I remember adding the `if` condition in `installerscript.qs` itself, so that it does not add the **Policy page**  in the first place. 
Let me know if this works

Reply:
@rish9511 Got it. Will take a look at this and explore silent updates in the next release.